File: web/src/lib/terminal-games.ts
export const TERMINAL_GAMES = [
  { id: "SDHD", label: "CHUNITHM" },
  { id: "SDEZ", label: "maimai DX" },
  { id: "SDGA", label: "maimai DX (Intl)" },
  { id: "SDED", label: "Card Maker" },
  { id: "SDDT", label: "O.N.G.E.K.I." },
  { id: "SBZV", label: "Project DIVA" },
  { id: "SDFE", label: "Wacca" },
] as const

export const DEFAULT_TERMINAL_GAME_ID = "SDEZ"

export function terminalGameLabel(gameID: string) {
  return (
    TERMINAL_GAMES.find((game) => game.id === gameID)?.label || gameID
  )
}

export function isTerminalGameID(gameID: string) {
  return TERMINAL_GAMES.some((game) => game.id === gameID)
}

File: internal/handler/aime/terminal_games.go
package aime

import "strings"

var supportedTerminalGameIDs = map[string]struct{}{
	"SDHD": {},
	"SDEZ": {},
	"SDGA": {},
	"SDED": {},
	"SDDT": {},
	"SBZV": {},
	"SDFE": {},
}

func normalizeTerminalGameID(value string) (string, bool) {
	gameID := strings.ToUpper(strings.TrimSpace(value))
	_, supported := supportedTerminalGameIDs[gameID]
	return gameID, supported
}

func NormalizeTerminalGameID(value string) (string, bool) {
	return normalizeTerminalGameID(value)
}
